### ISSUE TITLE
Add a `dry-run` option to prisma generate

### ISSUE THREAD

**Issue by zorzella** [prisma/prisma#24383]
## Problem

I would like to be able to run `prisma generate --dry-run` to see the output it would print without it actually modifying the prisma files.

## Suggested solution

Add a `dry-run` flag.

**Comment by janpio** [comment]
What would that look like? Prisma Client has quite some files, which are often very long. I can't imagine that outputting those in the terminal would be useful, wouldn't it?

**Comment by zorzella** [comment]
No, I'm not saying I want to output the files it would generate to stdout. I want to see the output that `prisma generate` would have printed in the stdout.

This might be useful, for example, for me to write a test that fails if it seems something like this:

```
warn Versions of prisma@5.14.0 and @prisma/client@5.4.1 don't match.
This might lead to unexpected behavior.
Please make sure they have the same version.
```

**Comment by zorzella** [comment]
Yes, your run-of-the-mill "dry run" behavior.

**Comment by SevInf** [comment]
Is there any specific information from `prisma generate` output that you are interested in? 
Usually, `--dry-run` is used for preview of a destructive non-undoable action to verify that change is actually correct. `prisma generate` right now does not really output anything that would allow you to do it. Moreover, `generate` change is not really destructive - same schema + same prisma version will always produce the same output.

**Comment by zorzella** [comment]
As I said, this might be useful, for example, for me to write a test that fails if `prisma generate` says something like this:

```
warn Versions of prisma@5.14.0 and @prisma/client@5.4.1 don't match.
This might lead to unexpected behavior.
Please make sure they have the same version.
```
